spec/models/track_spec.rb

```diff
@@ -23,23 +24,155 @@
+          track = build(:track, start_date: 3.days.ago, end_date: 2.days.ago, program: @conference.program)
+          expect(track.valid?).to eq false
+          expect(track.errors[:end_date]).to eq ["can't be before the conference start date (#{1.day.ago.to_date})"]
+        end
+
+        it 'when the track\'s start date is after the conference\'s end date' do
+          track = build(:track, start_date: 3.days.from_now, end_date: 4.days.from_now, program: @conference.program)
+          expect(track.valid?).to eq false
+          expect(track.errors[:start_date]).to eq ["can't be after the conference end date (#{2.days.from_now.to_date})"]
+        end
+
+        it 'when the track\'s end date is after the conference\'s end date' do
+          track = build(:track, start_date: Date.today, end_date: 3.days.from_now, program: @conference.program)
+          expect(track.valid?).to eq false
+          expect(track.errors[:end_date]).to eq ["can't be after the conference end date (#{2.days.from_now.to_date})"]
+        end
+
+        it 'when the track\'s start date is after it\'s end date' do
+          track = build(:track, start_date: 1.day.from_now, end_date: 1.day.ago)
+          expect(track.valid?).to eq false
+          expect(track.errors[:start_date]).to eq ['can\'t be after the end date']
+        end
+      end
+    end
+
+    describe '#valid_room' do
+      before :each do
+        @conference = create(:conference)
+        @conference.venue = create(:venue, name: 'The venue')
+      end
+
+      context 'is valid' do
+        it 'when the track\'s room belongs to the venue of the conference' do
+          room = create(:room, venue: @conference.venue)
+          track = build(:track, :self_organized, state: 'accepted', program: @conference.program, room: room)
+          expect(track.valid?).to eq true
+        end
+      end
+
+      context 'is invalid' do
+        it 'when the track\'s room doesn\'t belong to the venue of the track\'s conference' do
+          other_conference = create(:conference)
+          other_conference.venue = create(:venue)
+          room = create(:room, venue: other_conference.venue)
+          track = build(:track, :self_organized, state: 'accepted', program: @conference.program, room: room)
+          expect(track.valid?).to eq false
+          expect(track.errors[:room]).to eq ['must be a room of The venue']
+        end
+      end
+    end
+  end
+
+  describe 'scope' do
+    describe '#confirmed' do
+      before :each do
+        @program = create(:program)
+      end
+
+      context 'includes' do
+        it 'when track is confirmed' do
+          confirmed_track = create(:track, state: 'confirmed', program: @program)
+          expect(@program.tracks.confirmed.include?(confirmed_track)).to eq true
+        end
+      end
+
+      context 'excludes' do
+        %w[new to_accept accepted to_reject rejected canceled withdrawn].each do |state|
+          it "when track is #{state.humanize}" do
```

Comment:
Perhaps we can rename the tests, so that it makes more sense to read it. Right now this will read:
`Track scope #confirmed excludes when track is accepted`